diff --git a/bundle/nginx-1.29.2/src/core/ngx_log.c b/bundle/nginx-1.29.2/src/core/ngx_log.c
--- a/bundle/nginx-1.29.2/src/core/ngx_log.c
+++ b/bundle/nginx-1.29.2/src/core/ngx_log.c
@@ -159,8 +159,12 @@
     debug_connection = (log->log_level & NGX_LOG_DEBUG_CONNECTION) != 0;
 
     while (log) {
-
+#if (NGX_HTTP_LUA_KONG)
+        if (ngx_http_lua_kong_get_dynamic_log_level(log->log_level) < level &&
+            !debug_connection) {
+#else
         if (log->log_level < level && !debug_connection) {
+#endif
             break;
         }
 
@@ -218,7 +222,11 @@
 {
     va_list  args;
 
+#if (NGX_HTTP_LUA_KONG)
+    if (ngx_http_lua_kong_get_dynamic_log_level(log->log_level) >= level) {
+#else
     if (log->log_level >= level) {
+#endif
         va_start(args, fmt);
         ngx_log_error_core(level, log, err, fmt, args);
         va_end(args);
diff --git a/bundle/nginx-1.29.2/src/core/ngx_log.h b/bundle/nginx-1.29.2/src/core/ngx_log.h
--- a/bundle/nginx-1.29.2/src/core/ngx_log.h
+++ b/bundle/nginx-1.29.2/src/core/ngx_log.h
@@ -75,6 +75,13 @@
     NGX_COMPAT_END
 };
 
+#if (NGX_HTTP_LUA_KONG)
+ngx_uint_t
+ngx_http_lua_kong_get_dynamic_log_level(ngx_uint_t current_log_level);
+#else
+#define ngx_http_lua_kong_get_dynamic_log_level(expr) (expr)
+#endif
+
 
 #define NGX_MAX_ERROR_STR   2048
 
@@ -86,13 +93,13 @@
 #define NGX_HAVE_VARIADIC_MACROS  1
 
 #define ngx_log_error(level, log, ...)                                        \
-    if ((log)->log_level >= level) ngx_log_error_core(level, log, __VA_ARGS__)
+    if (ngx_http_lua_kong_get_dynamic_log_level((log)->log_level) >= level) ngx_log_error_core(level, log, __VA_ARGS__)
 
 void ngx_log_error_core(ngx_uint_t level, ngx_log_t *log, ngx_err_t err,
     const char *fmt, ...);
 
 #define ngx_log_debug(level, log, ...)                                        \
-    if ((log)->log_level & level)                                             \
+    if (ngx_http_lua_kong_get_dynamic_log_level((log)->log_level) & level)    \
         ngx_log_error_core(NGX_LOG_DEBUG, log, __VA_ARGS__)
 
 /*********************************/
@@ -102,13 +109,13 @@
 #define NGX_HAVE_VARIADIC_MACROS  1
 
 #define ngx_log_error(level, log, args...)                                    \
-    if ((log)->log_level >= level) ngx_log_error_core(level, log, args)
+    if (ngx_http_lua_kong_get_dynamic_log_level((log)->log_level) >= level) ngx_log_error_core(level, log, args)
 
 void ngx_log_error_core(ngx_uint_t level, ngx_log_t *log, ngx_err_t err,
     const char *fmt, ...);
 
 #define ngx_log_debug(level, log, args...)                                    \
-    if ((log)->log_level & level)                                             \
+    if (ngx_http_lua_kong_get_dynamic_log_level((log)->log_level) & level)    \
         ngx_log_error_core(NGX_LOG_DEBUG, log, args)
 
 /*********************************/
